inst/include/dust/dust.hpp

```diff
@@ -521,6 +540,72 @@ class Dust {
     for (size_t i = 0; i < particles_.size(); ++i) {
       res[i] = particles_[i].compare_data(data[i / np], rng_.state(i));
     }
+    stale_device_ = true; // RNG use
+  }
+
+  std::vector<real_t> compare_data_device() {
+    std::vector<real_t> res;
```

Comment:
This requires a device refresh

Reply:
I think that's done on L573 when `compare_data_device(x,x)` is called from this function

Reply:
fixed in 30f9eb8